frontend/src/firebase/actions.ts

```diff
@@ -152,21 +139,51 @@ export const removeAllForks = (taskId: string): void => {
   batch.commit().then(ignore);
 };
 
+type EditType = 'EDITING_MASTER_TEMPLATE' | 'EDITING_ONE_TIME_TASK' | 'FORKING_MASTER_TEMPLATE';
+
 export const editTaskWithDiff = (
   taskId: string,
-  forMasterTemplate: boolean,
+  editType: EditType,
   { mainTaskEdits, subTaskCreations, subTaskEdits, subTaskDeletions }: Diff,
 ): void => {
   const batch = db().batch();
-  if (forMasterTemplate) {
+  let updateTaskId = taskId;
+  if (editType === 'FORKING_MASTER_TEMPLATE') {
+    const { tasks, subTasks } = store.getState();
+    const repeatingTask = tasks.get(taskId) as RepeatingTask;
+    (async () => {
+      const createdSubTasks = repeatingTask.children.map((child) => {
+        const subtask = subTasks.get(child);
+        const firebaseSubTask: FirestoreSubTask = mergeWithOwner(subtask);
+        const subtaskDoc = subTasksCollection().doc();
+        batch.set(subtaskDoc, firebaseSubTask);
+        return { ...subtask, id: subtaskDoc.id };
+      });
+      await batch.commit();
+      const subtaskIds = createdSubTasks.map(s => s.id);
+      const taskWithChildren = { ...repeatingTask, children: subtaskIds };
+      const firestoreTask: FirestoreTask = await createFirestoreObject('tasks', taskWithChildren);
+      const addedDoc = await tasksCollection().add(firestoreTask);
+      const newForkMetaData = {
+        forkId: addedDoc.id,
+        replaceDate: mainTaskEdits.date !== undefined ? mainTaskEdits.date : repeatingTask.date,
+      };
+      batch.update(tasksCollection().doc(taskId), {
```

Comment:
You cannot use batch again. The `batch` is already committed above. At this point, you should think of this as an already closed file and you cannot do further operations on it.